src/Microsoft.AspNetCore.Hosting.Abstractions/IHostingEnvironment.cs

```diff
@@ -12,7 +12,8 @@ public interface IHostingEnvironment
     {
         /// <summary>
         /// Gets or sets the name of the environment. This property is automatically set by the host to the value
-        /// of the "ASPNETCORE_ENVIRONMENT" environment variable.
+        /// of the "ASPNETCORE_ENVIRONMENT" environment variable but can be overridden with the command line argument
+        /// --environment
```

Comment:
That opt in is via the config system. ASPNETCORE_ app environment variables are added to config by default, but any other config source must be added manually. E.g. you can define environment via command line, config file, etc., but you have to tell WebHostBuilder to use those config sources.
If we were to provide any hint here, it would be "This property is automatically set by the host to the value of the "ASPNETCORE_ENVIRONMENT" variable, or "environment" as specified in any other configuration source".